enhancements/results-forwarding.md

```diff
@@ -0,0 +1,528 @@
+* `extra`: `{'resourceType': 'cluster'}`
+  - Additional data provided by the Compliance Operator or configured through the `ScanSetting`.
+
+### Implementation Details/Notes/Constraints
+
+#### Work Items
+
+The following descibes the work needed to implement this feature, listed in
+order of implementation.
+
+1. Implement `ScanSetting` configuration options and bump the version to
+   `v1alpha2`
+2. Implement an interface for forwarding `ComplianceCheckResults` and
+   `ComplianceRemediations`
+3. Create a gRPC forwarding implementation
+4. Implement an idempotent process that creates a globally-readable `ConfigMap`
+   with the cluster ID
+5. Add `aggregator` support to forward `ComplianceCheckResults` and
+   `ComplianceRemediations` using the forwarding interface
+6. Implement a webhook that converts `v1alpha1` `ScanSetting` objects to `v1alpha2`
+
+### Risks and Mitigations
+
+`v1alpha1` objects would be easily translated to `v1alpha2` objects, but not the
+other way around. This needs to be thoroughly documented and communicated.
+
+## Design Details
+
+### Open Questions
+
+1. If the configured provider for either the result forwarder or the evidence
+   persistor would be unavailable, do we error the scan entirely?
+
+### Test Plan
+
+The base case of generating CRDs and storing evidence in PVCs would be covered
+in our prow-based pre-existing CI.
+
+New cases with forwarders would be introduced in more light-weight testing
+environments. e.g. we could deploy a KinD cluster via a GitHub action and run
+[MinIO](https://min.io/) to test the s3 provider. We'd then need to provide a
+reference GRPC receiver for the test.
+
+### Upgrade / Downgrade Strategy
+
+Upgrade expectations:
+- Existing `ScanSettings` should simply work and be seamlessly translated to `v1alpha2`
+
+Downgrade expectations:
+- Compliance Operator, as it is today, does not provide Downgrade options. This
+  is not expected to change.
+
+### Version Skew Strategy (TODO)
+
+How will the component handle version skew with other components?
+What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI
+  or CNI may require updating that component before the kubelet.
+
+### Operational Aspects of API Extensions (TODO)
+
+Requires a network connection to forward results after each scan. The
+Compliance Operator should validate the endpoint URL and fail early if it is
+malformed. If the Compliance Operator cannot connect to the gRPC endpoint, it
+should retry and issue an alert.
```

Comment:
This seems to answer the question posed on line 436, right?

Reply:
Yeah - I think we could supply a configuration, through documentation, that tells users how to configure the compliance-operator to forward results and how to store them locally.

The safety net in that case would be that the results are saved somewhere in a persistent volume if the gRPC forwarding fails.